src/ray/raylet/worker_killing_policy.cc

```diff
@@ -60,6 +60,45 @@ RetriableLIFOWorkerKillingPolicy::SelectWorkerToKill(
   return sorted.front();
 }
 
+GroupByDepthWorkingKillingPolicy::GroupByDepthWorkingKillingPolicy() {}
+const std::shared_ptr<WorkerInterface>
+GroupByDepthWorkingKillingPolicy::SelectWorkerToKill(
+    const std::vector<std::shared_ptr<WorkerInterface>> &workers,
+    const MemoryMonitor &memory_monitor) const {
+  if (workers.empty()) {
+    RAY_LOG_EVERY_MS(INFO, 5000) << "Worker list is empty. Nothing can be killed";
+    return nullptr;
+  }
+
+  std::vector<std::shared_ptr<WorkerInterface>> sorted = workers;
+  std::sort(sorted.begin(),
+            sorted.end(),
+            [](std::shared_ptr<WorkerInterface> const &left,
+               std::shared_ptr<WorkerInterface> const &right) -> bool {
+              // Sort by time in descending order
+              return left->GetAssignedTaskTime() > right->GetAssignedTaskTime();
+            });
+
+  std::map<int, std::vector<std::shared_ptr<WorkerInterface>>> group_by_depth;
+  for (const auto& worker : sorted) {
+    int depth = worker->GetAssignedTask().GetTaskSpecification().GetDepth();
```

Comment:
why not implement this in the comparator?

i.e.
sort by depth desc
if depth is the same, sort by time desc